### PR TITLE
Add proper multi-instance handling

### PR DESCRIPTION
**Pull request by phroggster**
Win32 APIs are horrible, but sometimes, they're the only way to get work done. Application will now attempt to focus the original window if it gets launched twice. If we can't raise/focus that window, fallback to the good old messagebox prompt.

Feedback welcome.

### CODE REVIEW

File: EDDiscovery/Program.cs [hunk 1]
```diff
@@ -31,15 +63,40 @@ static void Main()
...
+        /// <returns><c>true</c> if the other application windw was raised. <c>false</c> otherwise</returns>
+        static bool RaiseOtherProcess()
+        {
+            bool retval = false;
+            Process proc = Process.GetCurrentProcess();
+            foreach (Process otherProc in Process.GetProcessesByName(proc.ProcessName))
+            {
+                if (proc.Id != otherProc.Id)
+                {
+                    IntPtr hWnd = otherProc.MainWindowHandle;
```
Comment (klightspeed) [inline EDDiscovery/Program.cs]: A portable alternative might be:
```
Microsoft.VisualBasic.Interaction.AppActivate(otherProc.Id);
```
Though that won't restore the app if it's minimized.

This block could be wrapped in `try { } catch (DllNotFoundException) { } catch { }`, where the portable function is called if binding the Win32 functions fail, and `retval` is set to `false` if any other exception is caught.

**Comment by finwen** [comment]
Will test it litte later and want to know  klightspeed's thought about it with Mono build.

**Comment by phroggster** [comment]
This'll make Mono crash and burn faster than if it was being driven by a Kerbal. I'll submit some fixes in short order, soon as I can figure out whether #417 will need any of this.

**Comment by finwen** [comment]
Good.  I think i wait to merge this until we start working on 5.3 (soon)

**Comment by phroggster** [comment]
Sure enough, with the feature417 changeset applied, this approach falls to garbage. Going to have to go at the IPC from a different angle.

**Comment by klightspeed** [comment]
You would probably need something like a named pipe (`System.IO.Pipes.NamedPipeServerStream` / `System.IO.Pipes.NamedPipeClientStream`, which should be cross-platform) and/or a custom window message (processed in `EDDiscovery.EDDiscoveryForm.WindowProc`, which will only work on Windows).

**Comment by phroggster** [comment]
Closing this PR to get it out of the way. Bigger (and much better) solution coming in short order.

**Pull request closed without merging**